src/base/searchengine.h

```diff
@@ -38,10 +40,133 @@
 class QProcess;
 class QTimer;
 
+template <class T, std::size_t N>
+class Version
+{
+public:
+    typedef T ComponentType;
+    typedef Version<T,N> this_type;
+
+    Version()
+    {
+        m_components.fill(T());
+    }
+
+    template <class ...Other>
+    Version(Other... components)
+        : m_components {{components...}}
+    {
+    }
+
+    Version(const QString &version)
```

Comment:
From my point of view, a possibility to write a  version as a text is an inherent property of it. That is why I added those constructors. Following this path, `operator QString` and `operator QByteArray` may be of use. Your vision limits a version to an ordered number sequence which may be compared to another such sequence, as far as I understand. 

BTW, repeating the argument type in the  from- methods we just add unneeded noise, as to me. `fromString` would be enough.